src/CommonItilObject_Item.php

```diff
@@ -1578,4 +1509,213 @@ public static function dropdownAllDevices(
+     *
+     * @return void
+     **/
+    public static function itemAddForm(CommonDBTM $itil_object, $options = [])
+    {
+        $params = [
+            'id'  => (isset($itil_object->fields['id']) && $itil_object->fields['id'] != '') ? $itil_object->fields['id'] : 0,
+            'entities_id'  => (isset($itil_object->fields['entities_id']) && is_numeric($itil_object->fields['entities_id']) ? $itil_object->fields['entities_id'] : Session::getActiveEntity()),
+            '_users_id_requester' => 0,
+            'items_id'            => [],
+            'itemtype'            => '',
+            '_canupdate'          => false
+        ];
+
+        $opt = [];
+
+        foreach ($options as $key => $val) {
+            if (!empty($val)) {
+                $params[$key] = $val;
+            }
+        }
+
+        if (!$itil_object->can($params['id'], READ)) {
+            return false;
+        }
+
+        $can_add_items = $_SESSION["glpiactiveprofile"]["helpdesk_hardware"] & pow(2, $itil_object::HELPDESK_MY_HARDWARE) || $_SESSION["glpiactiveprofile"]["helpdesk_hardware"] & pow(2, $itil_object::HELPDESK_ALL_HARDWARE);
+        $canedit = ($can_add_items && $itil_object->can($params['id'], UPDATE)
+                  && $params['_canupdate']);
+
+       // Ticket update case
+        $usedcount = 0;
+        if ($params['id'] > 0) {
+           // Get requester
+            $class        = new $itil_object->userlinkclass();
+            $tickets_user = $class->getActors($params['id']);
+            if (
+                isset($tickets_user[CommonITILActor::REQUESTER])
+                && (count($tickets_user[CommonITILActor::REQUESTER]) == 1)
+            ) {
+                foreach ($tickets_user[CommonITILActor::REQUESTER] as $user_id_single) {
+                    $params['_users_id_requester'] = $user_id_single['users_id'];
+                }
+            }
+
+           // Get associated elements for ticket
+            $used = self::getUsedItems($params['id'], $itil_object->getType());
+            foreach ($used as $itemtype => $items) {
+                foreach ($items as $items_id) {
+                    if (
+                        !isset($params['items_id'][$itemtype])
+                        || !in_array($items_id, $params['items_id'][$itemtype])
+                    ) {
+                        $params['items_id'][$itemtype][] = $items_id;
+                    }
+                    ++$usedcount;
+                }
+            }
+        }
+
+        $rand  = mt_rand();
+        $count = 0;
+
+        $twig_params = [
+            'rand'               => $rand,
+            'can_edit'           => $canedit,
+            'my_items_dropdown'  => '',
+            'all_items_dropdown' => '',
+            'items_to_add'       => [],
+            'params'             => $params,
+            'opt'                => []
+        ];
+
+       // Get ticket template
+        $tt = new TicketTemplate();
```

Comment:
Needs made generic to handle other ITIL types?

Reply:
done